client2/src/game/Game.js

```diff
@@ -148,6 +150,26 @@ class Game extends Phaser.Scene {
          * @type {Object}
          */
         this.lightSources = {};
+
+        /**
+         * ChatInput status, determines whether chatInput is closed or opened.
+         * @type {Boolean}
+         */
+        this.chatInputStatus = false;
```

Comment:
I think it would be better to add this state to the GUI state instead of here, as it isn't gameplay/Phaser related as such https://github.com/Arcanorum/dungeonz/blob/feat/react/client2/src/shared/state/GUI.js

And then you can just reference it from there
`if (GUIState.chatInputStatus) return true;`
below

Reply:
Had I not [forgot](https://github.com/Arcanorum/dungeonz/pull/99#discussion_r570346707) about the GUI state, this would be there already.

Reply:
Refactored at 7ad4e8c